src/pybind/mgr/cephadm/module.py

```diff
@@ -1956,15 +1960,26 @@ def _create_osd(self, host, cmd):
             'keyring': keyring,
         })
 
-        before_osd_uuid_map = self.get_osd_uuid_map()
+        before_osd_uuid_map = self.get_osd_uuid_map(only_up=True)
 
         split_cmd = cmd.split(' ')
         _cmd = ['--config-and-keyring', '-', '--']
         _cmd.extend(split_cmd)
         out, err, code = self._run_cephadm(
             host, 'osd', 'ceph-volume',
             _cmd,
-            stdin=j)
+            stdin=j,
+            error_ok=True)
+        if code == 1 and ', it is already prepared' in '\n'.join(err):
+            # HACK: when we create against an existing LV, ceph-volume
+            # returns an error and the above message.  To make this
+            # command idempotent, tolerate this "error" and continue.
+            self.log.debug('the device was already prepared; continuing')
+            code = 0
```

Comment:
see https://tracker.ceph.com/issues/44824

Reply:
@jan--f can we extend this kludge to also cover https://tracker.ceph.com/issues/44824 ?

maybe something like ```python
'Calculated size of logical volume is 0 extents.' in err?
```